corehq/ex-submodules/casexml/apps/phone/tests/performance_tests.py

```diff
@@ -0,0 +1,161 @@
+                               password="changeme", date_joined=datetime(2011, 6, 9),
+                               additional_owner_ids=[REFERRED_TO_GROUP])
+
+        # this creates the initial blank sync token in the database
+        self.other_sync_log = synclog_from_restore_payload(generate_restore_payload(self.other_user))
+        self.referral_sync_log = synclog_from_restore_payload(generate_restore_payload(self.referral_user))
+
+        self.assertTrue(SHARED_ID in self.other_sync_log.owner_ids_on_phone)
+        self.assertTrue(OTHER_USER_ID in self.other_sync_log.owner_ids_on_phone)
+
+        self.user.additional_owner_ids = [SHARED_ID]
+        self.sync_log = synclog_from_restore_payload(generate_restore_payload(self.user))
+        self.assertTrue(SHARED_ID in self.sync_log.owner_ids_on_phone)
+        self.assertTrue(USER_ID in self.sync_log.owner_ids_on_phone)
+
+    @line_profile([
+        RestoreConfig.get_payload,
+        get_case_payload,
+        get_case_payload_batched,
+        get_related_cases,
+        filter_cases_modified_elsewhere_since_sync
+    ])
+    def test_profile_filter_cases_modified_elsewhere_since_sync(self):
+        total_cases = 100
+        proportion_modified = 0
+
+        modified = total_cases * proportion_modified
+        id_list = ['case_id_{}'.format(i) for i in range(total_cases)]
+        self._createCaseStubs(id_list, user_id=USER_ID, owner_id=SHARED_ID)
+
+        caseblocks = []
+        for case_id in id_list[:modified]:
+            caseblocks.append(CaseBlock(
+                case_id=case_id,
+                user_id=OTHER_USER_ID,
+                version=V2,
+                update={'favorite_color': 'blue'}
+            ).as_xml())
+        self._postFakeWithSyncToken(caseblocks, self.other_sync_log.get_id)
+
+        assert_user_has_cases(self, self.user, id_list[:modified], restore_id=self.sync_log.get_id)
+
+    @line_profile([
+        RestoreConfig.get_payload,
+        get_case_payload,
+        get_case_payload_batched,
+        get_related_cases,
+        filter_cases_modified_elsewhere_since_sync
+    ])
+    def test_profile_get_related_cases(self):
+        total_parent_cases = 50
+
+        id_list = ['case_id_{}'.format(i) for i in range(total_parent_cases)]
+        self._createCaseStubs(id_list, user_id=USER_ID, owner_id=SHARED_ID)
+
+        new_case_ids = []
+        caseblocks = []
+        for i, parent_case_id in enumerate(id_list):
+            case_id = 'case_id_referral_{}'.format(i)
+            new_case_ids.append(case_id)
+            caseblocks.append(CaseBlock(
+                create=True,
+                case_id=case_id,
+                user_id=USER_ID,
+                owner_id=REFERRED_TO_GROUP,
+                case_type=REFERRAL_TYPE,
+                version=V2,
+                index={'parent': (PARENT_TYPE, parent_case_id)}
+            ).as_xml())
+        self._postFakeWithSyncToken(caseblocks, self.sync_log.get_id)
```

Comment:
Nothing jumps out but happy to take a look. I have to imagine we support creating a parent and children in the same form.... right? Would be crazy if that's never come up till now.